RFC/src/RFC-0140 Syncing and seeding.md

```diff
@@ -0,0 +1,132 @@
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This document describes the process of Syncing, seeding, pruning and cut-through.
+
+## Related RFCs
+
+* [RFC-0110: Base Nodes](RFC-0110_BaseNodes.md)
+
+## Descriptions
+
+### Syncing
+
+When a new node comes online, loses connection or encounters a chain re-organisation that is longer than it can tolerate, it must enter syncing mode. This will allow it to recover its state to the newest up to date state. Syncing can be divided into 2 [SynchronisationStrategy]s, complete sync and sync. Complete sync will involve that the node communicates with an archive node to get the complete history of every single block from genesis block. Sync will involve the node getting every block from its [pruning horizon](pruninghorizon) to [current head](current head), as well as every block header from genesis block. 
+
+#### Complete Sync
+
+Complete sync is only available from archive nodes, as these will be the only nodes that will be able to supply the complete history required to sync every block with every transaction from genesis block up onto [current head](currenthead). 
+
+
+
+#### Syncing process
+
+The syncing process MUST be done in the following steps:
+
+1. Set its [SynchronisationState] to `Synchronising`.
+2. Load a bootstrap list of peers from a configuration file, or a cached list.
+3. For each peer in this list:
+   1. Establish a connection with the peer.
+   2. Request a peer list from that peer.
+   3. Request information about the most recent chain state (total accumulated work, block height, etc.) from the peer.
+4. Choose the longest chain based on POW. 
+5. Download all headers from genesis block up onto [current head](currenthead), and validate the headers as you receive them.
+6. Download [UTXO](utxo) set at [pruning horizon](pruninghorizon). 
+7. Download all blocks from  [pruning horizon](pruninghorizon) up onto [current head](currenthead), if the node is doing a complete sync, the [pruning horizon](pruninghorizon) will just be infinite, which means you will download all blocks ever created.
+8. Validate blocks as if they where just mined and then received, in chronological order. 
+
+After this process the node will be in sync and able to process blocks and transaction normally. 
+
+#### Keeping in sync
+
+The node SHOULD periodically test its peers with ping messages to ensure that they are alive. When a node sends a ping message, it should include the current total POW, hash of the [current head](currenthead) and genesis block hash of its own current longest chain in the ping message. The receiving node MUST relay with a pong message also including the total POW, [current head](currenthead) and genesis block hash of its longest chain. If the two chains dont match up, the node with the lowest POW has the responsiblity to ask the peer for syncing information. 
```

Comment:
grammar: "The receiving node MUST `reply` with a pong message including their own total PoW..."